org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/Preferences.java

```diff
@@ -1330,9 +1336,36 @@ public static Preferences createFrom(Map<String, Object> configuration) {
 		prefs.setChainCompletionEnabled(chainCompletionEnabled);
 		List<String> diagnosticFilter = getList(configuration, JAVA_DIAGNOSTIC_FILER, Collections.emptyList());
 		prefs.setDiagnosticFilter(diagnosticFilter);
+		Object object = getValue(configuration, JAVA_CONFIGURATION_ASSOCIATIONS);
+		Set<String> associations = new HashSet<>();
+		if (object instanceof Map map) {
+			try {
+				Map<String, String> element = map;
+				element.forEach((k, v) -> {
+					// Java LS only support a small subset of the glob pattern syntax (*.xxx)
+					if ("java".equals(v) && validateFileName(k)) {
+						associations.add(k.substring(2));
+					}
+				});
+			} catch (Exception e) {
+				JavaLanguageServerPlugin.logException(e);
+			}
+		}
+		prefs.setFilesAssociations(new ArrayList<>(associations));
 		return prefs;
 	}
 
+	private static boolean validateFileName(String filename) {
```

Comment:
I would call this `validateFilePattern`.

Reply:
Fixed.